pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/IngestionUtils.java

```diff
@@ -169,6 +170,16 @@ private static SegmentNameGenerator getSegmentNameGenerator(BatchConfig batchCon
       case BatchConfigProperties.SegmentNameGeneratorType.SIMPLE:
         return new SimpleSegmentNameGenerator(rawTableName, batchConfig.getSegmentNamePostfix(),
             batchConfig.isAppendUUIDToSegmentName(), batchConfig.isExcludeTimeInSegmentName());
+      case BatchConfigProperties.SegmentNameGeneratorType.UPLOADED_REALTIME:
+        int uploadedRealtimePartitionId;
+        try {
+          uploadedRealtimePartitionId = Integer.parseInt(batchConfig.getUploadedRealtimePartitionId());
+        } catch (NumberFormatException e) {
+          throw new IllegalArgumentException(
+              String.format("Invalid uploadedRealtimePartitionId: %s", batchConfig.getUploadedRealtimePartitionId()));
+        }
+        return new UploadedRealtimeSegmentNameGenerator(rawTableName, uploadedRealtimePartitionId,
+            batchConfig.getSegmentUploadTimeMs(), batchConfig.getSegmentNamePrefix(), batchConfig.getSequenceId());
```

Comment:
how many segments are generated by one flink task? I wonder if the sequentId should be adjusted to avoid overwriting segments uploaded by one task.

Reply:
The segment count can be unbounded. The sequenceId gets generated per task i.e. if parallelism is 4, there will be 4 tasks each with their FlinkSegmentWriter instance and sequenceId gets incremented after each segment flush and upload.

Given sequenceId is incremental and partitionId differentiates each segment name, there isn't a possibility of overwrite by another task's same sequenceId.

Reply:
I see, and I just spotted this in the README that each task will generate segments for one particular table partition, then it make sense to use a sequenceId local to each task.
```
The parallelism of the job *must* be set the same as the number of partitions of the Pinot table(same as upstream), so that the sink in each task executor can generate the segment of same partitions.
```